### PR TITLE
Mve2 pong api

### PR DESCRIPTION
**Pull request by romann-broque**

### CODE REVIEW

File: back/transcendence_django/back_game/game_physics/edges.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+from back_game.game_physics.position import Position
+
+class Edges:
+    def __init__(self, position: Position, width: int, height: int):
+        self.update(position, width, height)
+
+    def update(self, position: Position, width: int, height: int):
+        self.bottom: float = position.y + height / 2
```
Comment (romann-broque) [inline back/transcendence_django/back_game/game_physics/edges.py:8]: initialize attributes in the __init__ method and move the types from the update method to the __init__

File: back/transcendence_django/back_game/game_physics/position.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class Position:
+    def __init__(self, x:float=0, y:float=0):
+        self.x = x
+        self.y = y
```
Comment (romann-broque) [inline back/transcendence_django/back_game/game_physics/position.py:4]: types

File: back/transcendence_django/back_game/game_physics/rectangle.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            "height": self.height,
+        }
+
+    def update_position(self, position: Position):
+        self.position = position
+        self.edges.update(position, self.width, self.height)
+        self.convexity_center = self.__get_convexity_center()
+
+    def __get_convexity_center(self) -> Position:
+        self.distance_from_center = self.height / (2 * math.tan(CONVEXITY / 2))
```
Comment (romann-broque) [inline back/transcendence_django/back_game/game_physics/rectangle.py:31]: Store the value 2 * math.tan(CONVEXITY / 2) into a constant TANGENT_FACTOR

File: back/transcendence_django/back_game/monitor/monitor.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    RUN_LOOP_INTERVAL,
+    STARTED,
+    TIMEOUT_GAME_OVER,
+    TIMEOUT_INTERVAL,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class Monitor:
```
Comment (romann-broque) [inline back/transcendence_django/back_game/monitor/monitor.py:21]: add key dictionnary constants

File: front/src/app/components/gameover/gameover.component.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  public backToHomePage() {
+    this.webSocketService.giveUp();
+    this.redirectToHome();
+  }
+
+  public redirectToHome() {
+    this.router.navigate(['/home']);
+  }
+
+  reMatch() {
```
Comment (romann-broque) [inline front/src/app/components/gameover/gameover.component.ts:33]: public

File: front/src/app/components/paddle/paddle.component.ts [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+  imports: [
+    NgIf
+  ],
+  templateUrl: './paddle.component.html',
+  styleUrl: './paddle.component.css'
+})
+export class PaddleComponent implements OnInit {
+  @Input() id: number = 0;
+  playerName: string = '';
+  width = PADDLE_WIDTH;
```
Comment (romann-broque) [inline front/src/app/components/paddle/paddle.component.ts:18]: types !

File: front/src/app/constants.ts [hunk 7]
```diff
@@ -0,0 +1,43 @@
+// constants.ts
+
+////////////////
+// COMPONENTS //
+////////////////
+
+export const LOADING_BUTTON_TIME = 5000; // 5000 ms = 5 s
```
Comment (romann-broque) [inline front/src/app/constants.ts:7]: remove this variable

File: front/src/app/pages/game-page/game-page.component.ts [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+  @HostListener('window:resize', ['$event'])
+  public onResize(event: Event) {
+    this.updateGameContainerScale();
+  }
+
+  constructor(
+    private elementRef: ElementRef,
+    private router: Router,
+    private webSocketService: WebSocketService,
+    private route: ActivatedRoute,  // Inject ActivatedRoute
```
Comment (romann-broque) [inline front/src/app/pages/game-page/game-page.component.ts:43]: remove comments

File: front/src/app/pages/game-page/game-page.component.ts [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+    this.connectionService.endConnection();
+  }
+
+  private updateGameContainerScale() {
+    const gameContainer = this.elementRef.nativeElement.querySelector('.game-container');
+    const scale = Math.min(window.innerWidth / 1000, window.innerHeight / 1000);
+    gameContainer.style.transform = `scale(${scale})`;
+  }
+
+  confirmGiveUp() {
```
Comment (romann-broque) [inline front/src/app/pages/game-page/game-page.component.ts:71]: public

File: front/src/app/components/loading-spinner/loading-spinner.component.css [hunk 10]
```diff
@@ -0,0 +1,40 @@
+.spinner {
+  width: 40px;
```
Comment (opaleboreale) [inline front/src/app/components/loading-spinner/loading-spinner.component.css:2]: px -> %

**Review by romann-broque — COMMENTED** [review]
not bad! :D

**Review by romann-broque — COMMENTED** [review]
Nice!

**Review by opaleboreale — APPROVED** [review]

**Pull request merged**